trio/_channel.py

```diff
@@ -306,15 +361,19 @@ async def receive(self):
         self._state.receive_tasks[task] = None
         task.custom_sleep_data = self
 
-        def abort_fn(_):
+        def abort_fn(_: RaiseCancelT) -> Abort:
             self._tasks.remove(task)
             del self._state.receive_tasks[task]
             return trio.lowlevel.Abort.SUCCEEDED
 
-        return await trio.lowlevel.wait_task_rescheduled(abort_fn)
+        result = await trio.lowlevel.wait_task_rescheduled(abort_fn)
+
+        # wait_task_rescheduled is set to return Any currently, but maybe it's
+        # guaranteed to return ReceiveType (or get aborted)?
+        return result  # type: ignore
```

Comment:
Looking at the code: it's not *guaranteed* to return `ReceiveType` (you could always call `reschedule` yourself and ruin trio's day...) but it's close enough: . I think the comment should be updated the reflect this, but having the `type: ignore` is aok. (Also I misunderstood earlier -- we don't need to move the `return` to a new line, I had thought `wait_task_rescheduled` just wouldn't return)

https://github.com/python-trio/trio/blob/ca850f67c84c06e79a8d5ae6ed7b24a000e8d024/trio/_channel.py#L146